### ISSUE TITLE
drain.Helper defaults to "delete immediately"

### ISSUE THREAD

**Issue by bboreham** [kubernetes/kubectl#982]
The member `GracePeriodSeconds` defaults to 0, which is interpreted as "delete immediately" by the lower layers called.
I'm not sure whether this is a bug or some subtle design I haven't figure out.

IMHO it would have been better to copy `metav1.DeleteOptions` which makes `GracePeriodSeconds` a pointer, so the default value is nil.

**Comment by michaelgugino** [comment]
When used via kubectl cli, this defaults to -1 as expected: https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/drain/drain.go#L148

When used as a library, the value should be specified or standard go defaults apply.

**Comment by bboreham** [comment]
Yes, I'm saying the behaviour when used as a library should be different, because the current behaviour is extremely surprising.
Not at a Go level, at a "all your pods just got killed instantly" level.

**Comment by dougsland** [comment]
@bboreham the patch https://github.com/kubernetes/kubernetes/pull/96812 is merged. Can we close this one?

**Comment by dougsland** [comment]
> Yes, I'm saying the behaviour when used as a library should be different, because the current behaviour is extremely surprising.
> Not at a Go level, at a "all your pods just got killed instantly" level.

My main concern is, changing behavior as a library, might break users out there. Let me know your thoughts @bboreham

**Comment by bboreham** [comment]
My thoughts are the current behaviour is broken. 

Changing to a pointer will silently fix anyone who didn’t realise, and give a small roadbump to those who did.

**Comment by dougsland** [comment]
let me take a deep look in this one.
/assign

**Comment by bboreham** [comment]
Example where I had to fix this: https://github.com/weaveworks/kured/pull/237

**Comment by dougsland** [comment]
Sorry the delay, this week I will get back to this issue.
